snorkel/analysis/utils.py

```diff
@@ -49,71 +46,44 @@ def preds_to_probs(preds: np.ndarray, num_classes: int) -> np.ndarray:
     return np.eye(num_classes)[preds.squeeze() - 1]
 
 
-def arraylike_to_numpy(
-    array_like: ArrayLike, flatten: bool = True, cast_to_int: bool = True
-) -> np.ndarray:
-    """Convert an ArrayLike (e.g., list, tensor, etc.) to a numpy array.
+def to_int_label_array(X: np.ndarray, flatten_vector: bool = True) -> np.ndarray:
+    """Convert an array to a (possibly flattened) array of ints.
 
-    Also optionally flatten [n, 1] arrays to [n] and cast all values to ints.
+    Cast all values to ints and possibly flatten [n, 1] arrays to [n].
     This method is typically used to sanitize labels before use with analysis tools or
-    metrics that expect 1D numpy arrays as inputs.
+    metrics that expect 1D arrays as inputs.
 
     Parameters
     ----------
-    array_like
-        An Arraylike to convert
-    flatten
-        If True, flatten numpy array into a 1D array
-    cast_to_int
-        If True, cast all values to ints
+    X
+        An array to possibly flatten and possibly cast to int
+    flatten_vector
+        If True, flatten array into a 1D array
 
     Returns
     -------
     np.ndarray
-        The input converted to an np.ndarray
+        The converted array
 
     Raises
     ------
     ValueError
         Provided input could not be converted to an np.ndarray
     """
-
-    orig_type = type(array_like)
-
-    # Convert to np.ndarray
-    if isinstance(array_like, np.ndarray):
-        pass
-    elif isinstance(array_like, list):
-        array_like = np.array(array_like)
-    elif isinstance(array_like, sparse.spmatrix):
-        array_like = array_like.toarray()
-    elif isinstance(array_like, torch.Tensor):
-        array_like = array_like.numpy()
-    elif not isinstance(array_like, np.ndarray):
-        array_like = np.array(array_like)
-    else:
-        msg = f"Input of type {orig_type} could not be converted to an np.ndarray"
-        raise ValueError(msg)
-
+    if np.any(np.not_equal(np.mod(X, 1), 0)):
```

Comment:
Can we just check the type of the numpy array instead?

Reply:
isn't the point that we cast to int if it's a float array but they're all actually integers?